snewpdag/plugins/ValidateListType.py

```diff
@@ -0,0 +1,69 @@
+'''
+This will check the types of the elements in a list,
+and will consume the action if
+more than 10% of the data are not the right type.
+Then we don't have to have each downstream plugin
+repeat this sort of validation.
+
+check_listtype() checks the type of the elements in the list
+based on the type we want in the input,
+and if a wrong type is found, such element is removed
+and the removal is logged.
+
+min_fraction is the maximum fraction of numbers with wrong key_type allowed
+that we don't need to consume the action.
+e.g. min_fraction = 0.1 (i.e. 10%)
+I. if there are 10% or fewer elements in the list with wrong key type,
+we simply remove the element and log the removal.
+II. if there are more than 10% with wrong key type, we consume the action.
+
+Input: data as payload + min_fraction + key_type
+'''
+
+import logging
+from snewpdag.dag import Node
+
+class ValidateListType(Node):
+    def __init__(self, max_fraction, **kwargs):
+        self.max_fraction = max_fraction
+        self.key_type = kwargs.pop('key_type', None)
```

Comment:
It seems to me key_type should also be a required argument.  kwargs.pop() is used for optional ones.